UITests/BookmarksPanelTests.swift

```diff
@@ -144,4 +152,36 @@ class BookmarksPanelTests: KIFTestCase {
         navigateFolder(withTitle: "remote-subfolder")
         assertRowExists(withTitle: "item-in-remote-subfolder")
     }
+
+    // This is a continuation of the above test
+    func verifyMobileBookmarkDelete() {
+        // Assert precondition that this test continues from previous state
+        assertRowExists(withTitle: "item-in-remote-subfolder")
+
+        let deleteAction = GREYMatchers.matcher(forText: "Remove Bookmark")
+
+        // Test deletion not available
+
+        var row = EarlGrey.select(elementWithMatcher: grey_accessibilityLabel("item-in-remote-subfolder"))
+        row.perform(grey_longPress())
+        // verify the context menu does not have deletion option
+        EarlGrey.select(elementWithMatcher: deleteAction).assert(grey_nil())
+        // close the context menu
+        EarlGrey.select(elementWithMatcher: GREYMatchers.matcher(forText: "Pin to Top Sites")).perform(grey_tap())
+
+        navigateBackInTableView()
+
+        // Test successful deletion
+
+        assertRowExists(withTitle: "xyz")
+
+        let tv = tester().waitForView(withAccessibilityIdentifier: "Bookmarks List") as! UITableView
```

Comment:
I don't see how to do this in EarlGrey. Very easy to do in KIF of course.

Reply:
this is good enough, we can create expectations in earl grey, but it'll get pretty long